histomicstk/cli/ColorDeconvolution/ColorDeconvolution.xml

```diff
@@ -25,26 +25,25 @@
       <longflag>region</longflag>
       <default>-1,-1,-1,-1</default>
     </region>
-    <image fileExtensions=".png" reference="inputImageFile">
+    <image fileExtensions=".tiff" reference="inputImageFile">
       <name>outputStainImageFile_1</name>
       <label>Output Image of Stain 1</label>
-      <description>Output Image of Stain 1 (*.png)</description>
+      <description>Output Image of Stain 1 (*.tiff)</description>
       <channel>output</channel>
       <index>1</index>
     </image>
-    <image fileExtensions=".png" reference="inputImageFile">
+    <image fileExtensions=".tiff" reference="inputImageFile">
       <name>outputStainImageFile_2</name>
       <label>Output Image of Stain 2</label>
-      <description>Output Image of Stain 2 (*.png)</description>
+      <description>Output Image of Stain 2 (*.tiff)</description>
       <channel>output</channel>
       <index>2</index>
     </image>
-    <image fileExtensions=".png" reference="inputImageFile">
+    <image fileExtensions=".tiff" reference="inputImageFile">
       <name>outputStainImageFile_3</name>
       <label>Output Image of Stain 3</label>
-      <description>Output Image of Stain 3 (*.png)</description>
+      <description>Output Image of Stain 3 (*.tiff)</description>
       <channel>output</channel>
-      <index>3</index>
```

Comment:
I would expect removing the index parameter to fail some sort of validation.  This is officially specifying that the CLI takes positional inputs (e.g., `<command> <inputImageFile> <outputStainImageFile_1> <outputStainImageFile_2> <outputStainImageFile_3>`).  Without the `<index>3</index>` parameter and no flags, there is no way to specify the third output file.

The actual issue is that I had said not to emit the second and/or third stain image files unless they would be used.  We need to put that back.  Better (but not this PR) would be to add a flag in the slicer_cli_web extensions to ctk_cli that would allow for optional output files.  girder_worker via slicer_cli_web throws an exception is we were told such a file should exist and it doesn't.

Reply:
> I would expect removing the index parameter to fail some sort of validation. This is officially specifying that the CLI takes positional inputs (e.g., `<command> <inputImageFile> <outputStainImageFile_1> <outputStainImageFile_2> <outputStainImageFile_3>`). Without the `<index>3</index>` parameter and no flags, there is no way to specify the third output file.
> 
> The actual issue is that I had said not to emit the second and/or third stain image files unless they would be used. We need to put that back. Better (but not this PR) would be to add a flag in the slicer_cli_web extensions to ctk_cli that would allow for optional output files. girder_worker via slicer_cli_web throws an exception is we were told such a file should exist and it doesn't.

Okay, So you are recommending removing the last commit which adds these changes, and on another PR add a flag in the `slicer_cli_web` extensions.

Reply:
Until we make a change in some upstream library (probably slicer_cli_web), we have to leave the index parameter in the xml file AND emit the images files for all three stains.